lib/navigator/main_navigator_state.dart

```diff
@@ -0,0 +1,82 @@
+part of 'main_navigator.dart';
+
+/// The MainNavigatorState class is the state associated with the MainNavigator
+/// widget. It includes methods and properties for managing navigation within
+/// the application.
+class MainNavigatorState extends State<MainNavigator> with MainNavigationMixin {
+  static final GlobalKey<NavigatorState> _navigationKey =
+      GlobalKey<NavigatorState>();
+  static final List<NavigatorObserver> _navigatorObservers = [];
+
+  /// The initial route for the navigator.
+  static String get initialRoute => RouteNames.splash;
+
+  /// A global key for accessing the navigator state.
+  static GlobalKey<NavigatorState> get navigationKey => _navigationKey;
+
+  /// A list of navigator observers for monitoring navigation events.
+  static List<NavigatorObserver> get navigatorObservers => _navigatorObservers;
+
+  /// A getter for accessing the current state of the navigator.
+  NavigatorState get navigator => _navigationKey.currentState!;
+
+  @override
+  Widget build(BuildContext context) {
+    return TextScaleFactor(
+      child: widget.child ?? const SizedBox.shrink(),
+    );
+  }
+
+  /// A static method for generating routes based on the given [settings]. It
+  /// maps route names to the corresponding widget builders.
+  ///
+  /// If the route name does not match any of the defined routes, it defaults
+  /// to the SplashScreen.
+  ///
+  /// Example route settings:
+  /// ```dart
+  /// RouteSettings(name: RouteNames.login)
+  /// ```
+  static Route? onGenerateRoute(RouteSettings settings) {
```

Comment:
Nice!

We might end up using a routing package so not sure how much of this we'll keep though. But I think for now we can have this.